src/components/images/draw.js

```diff
@@ -142,8 +160,29 @@ module.exports = function draw(gd) {
         var sizing = anchors.x[d.xanchor].sizing + anchors.y[d.yanchor].sizing;
 
         // Final positions
-        var xPos = (xa ? xa.r2p(d.x) + xa._offset : d.x * size.w + size.l) + xOffset;
-        var yPos = (ya ? ya.r2p(d.y) + ya._offset : size.h - d.y * size.h + size.t) + yOffset;
+        var xPos, yPos;
+        if(xa !== undefined) {
+            if((typeof(d.xref) === 'string') && d.xref.endsWith(' domain')) {
+                xPos = xa._length * d.x + xa._offset;
+            } else {
+                xPos = xa.r2p(d.x) + xa._offset;
+            }
+        } else {
+            xPos = d.x * size.w + size.l;
+        }
```

Comment:
Also here let's try keeping those ternary operators to avoid `if/else` blocks.
```js
var xPos = !xa ? d.x * size.w + size.l :
  typeof(d.xref) === 'string' && d.xref.endsWith(' domain') ?
    xa._length * d.x + xa._offset :
    xa.r2p(d.x) + xa._offset;

xPos += xOffset;
```